### ISSUE TITLE
Correcting amplify cloudformation stack drift.

### ISSUE THREAD

**Issue by chrisfowler** [aws-amplify/amplify-cli#3810]
**Which Category is your question related to?**

`Drift` Detection in Runtime setting for `amplify` auto generated Lambda function (UserPoolClientLambda). 

Stack is in:
```
Status
UPDATE_ROLLBACK_FAILED
Status reason
The following resource(s) failed to update: [UserPoolClientLambda].
```
see further:
```
The runtime parameter of nodejs6.10 is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs12.x) while creating or updating functions. (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException; Request ID: bcb35e19-d746-4ddd-8bfa-1d7b6bbb5257)
```

**Amplify CLI Version** 

```
❯ amplify -v
4.14.1
```

**What AWS Services are you utilizing?**
amplify, cloudformation, lambda

**Provide additional details e.g. code snippets**
I have updated Cloudformation to set lambda runtime to nodejs12.x

The amplify deployment bucket contains the updated cloudformation.yml with the new runtime setting.

The lamda template viewed on the aws console shows the old setting (node6.10)

Running the Drift Detection I see this:
![Screenshot 2020-03-31 at 09 16 03](https://user-images.githubusercontent.com/51089/78003836-3059ba80-7331-11ea-91f8-42e6c261fe42.png)

I cannot change the lambda runtime back to node6.10 on the console as it is no longer supported and therefore not an option.

How can I update the Lambda template to reflect the new runtime: nodejs12.x? Trying to do `amplify auth push` from the CLI produces error due to: `is in UPDATE_ROLLBACK_FAILED state and can not be updated.`

This is my `staging` env, so would like some advice on best way to proceed. I will get the same issue in `prod`.

I WANT TO AVOID DELETING THE STACK!

**Comment by kaustavghosh06** [comment]
@chrisfowler You can update the Cloudformation templates located in `amplify/backend` directory of your project and run `amplify push`. You should treat these Cloudformation templates are your source of truth and would overwrite any changes you made on the AWS console.

**Comment by chrisfowler** [comment]
@kaustavghosh06 I did update amplify/backend cloudformation and did `amplify push` (see above) but is failing due to 'UPDATE_ROLLBACK_FAILED' status of stack.

Stack update is failing as existing template to the Lambda in question is asking for a runtime that is no longer supported: node6.10.

Can I update the existing template, so the roll-back would ask for node12.x ?

**Comment by chrisfowler** [comment]
Output below.

Because the rollout is failing, Cloudformation tries to rollback. Bu the Lambda function runtime is set to an unsupported node version (original deployment was over year ago). In short I am blocked to go forward, and blocked to rollback.

Where does `amplify pull` fetch the cloudformation from? Is this generated on the fly? 

```
~/dev/qi-ui-admin master* 10s
❯ amplify push --force
✔ Successfully pulled backend environment staging from the cloud.

Current Environment: staging

| Category | Resource name   | Operation | Provider plugin   |
| -------- | --------------- | --------- | ----------------- |
| Auth     | cognito62efae15 | Update    | awscloudformation |
| Api      | qiAppSync       | No Change | awscloudformation |
| Hosting  | S3AndCloudFront | No Change | awscloudformation |
? Are you sure you want to continue? Yes

...
⠏ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE    hostingS3AndCloudFront         AWS::CloudFormation::Stack Wed Apr 01 2020 13:35:27 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS hostingS3AndCloudFront         AWS::CloudFormation::Stack Wed Apr 01 2020 13:35:26 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS AuthRole                       AWS::IAM::Role             Wed Apr 01 2020 13:35:26 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS UnauthRole                     AWS::IAM::Role             Wed Apr 01 2020 13:35:26 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS qi-web-platform-xxxxxxxx AWS::CloudFormation::Stack Wed Apr 01 2020 13:35:22 GMT+0100 (British Summer Time) User Initiated
⠏ Updating resources in the cloud. This may take a few minutes...

UPDATE_IN_PROGRESS authcognito62efae15            AWS::CloudFormation::Stack Wed Apr 01 2020 13:35:43 GMT+0100 (British Summer Time)
CREATE_IN_PROGRESS UpdateRolesWithIDPFunctionRole AWS::IAM::Role             Wed Apr 01 2020 13:35:42 GMT+0100 (British Summer Time)
UPDATE_COMPLETE    AuthRole                       AWS::IAM::Role             Wed Apr 01 2020 13:35:41 GMT+0100 (British Summer Time)
UPDATE_COMPLETE    UnauthRole                     AWS::IAM::Role             Wed Apr 01 2020 13:35:40 GMT+0100 (British Summer Time)
⠴ Updating resources in the cloud. This may take a few minutes...

UPDATE_IN_PROGRESS qi-web-platform-xxxxxxx-authcognito62efae15-1FGX3HR4IUXCJ AWS::CloudFormation::Stack Wed Apr 01 2020 13:35:43 GMT+0100 (British Summer Time) User Initiated
⠇ Updating resources in the cloud. This may take a few minutes...

CREATE_IN_PROGRESS UpdateRolesWithIDPFunctionRole AWS::IAM::Role Wed Apr 01 2020 13:35:43 GMT+0100 (British Summer Time) Resource creation Initiated
⠴ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE    UserPool AWS::Cognito::UserPool Wed Apr 01 2020 13:35:51 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS UserPool AWS::Cognito::UserPool Wed Apr 01 2020 13:35:49 GMT+0100 (British Summer Time)
⠴ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE    UserPoolClientLambda AWS::Lambda::Function Wed Apr 01 2020 13:35:57 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS UserPoolClientLambda AWS::Lambda::Function Wed Apr 01 2020 13:35:57 GMT+0100 (British Summer Time)
⠇ Updating resources in the cloud. This may take a few minutes...

CREATE_COMPLETE UpdateRolesWithIDPFunctionRole AWS::IAM::Role Wed Apr 01 2020 13:35:59 GMT+0100 (British Summer Time)
⠦ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE_CLEANUP_IN_PROGRESS qi-web-platform-xxxxxx-authcognito62efae15-1FGX3HR4IUXCJ AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:09 GMT+0100 (British Summer Time)
⠹ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE authcognito62efae15 AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:17 GMT+0100 (British Summer Time)
⠙ Updating resources in the cloud. This may take a few minutes...

UPDATE_ROLLBACK_IN_PROGRESS qi-web-platform-xxxxxx AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:21 GMT+0100 (British Summer Time) The following resource(s) failed to update: [apiqiAppSync].
CREATE_COMPLETE             UpdateRolesWithIDPFunction     AWS::Lambda::Function      Wed Apr 01 2020 13:36:20 GMT+0100 (British Summer Time)
UPDATE_FAILED               apiqiAppSync                   AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:20 GMT+0100 (British Summer Time) Parameters: [CreateAPIKey] do not exist in the template
CREATE_IN_PROGRESS          UpdateRolesWithIDPFunction     AWS::Lambda::Function      Wed Apr 01 2020 13:36:19 GMT+0100 (British Summer Time) Resource creation Initiated
CREATE_IN_PROGRESS          UpdateRolesWithIDPFunction     AWS::Lambda::Function      Wed Apr 01 2020 13:36:19 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS          apiqiAppSync                   AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:19 GMT+0100 (British Summer Time)
⠹ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE    hostingS3AndCloudFront AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:29 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS hostingS3AndCloudFront AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:29 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS UnauthRole             AWS::IAM::Role             Wed Apr 01 2020 13:36:28 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS AuthRole               AWS::IAM::Role             Wed Apr 01 2020 13:36:28 GMT+0100 (British Summer Time)
⠙ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE AuthRole   AWS::IAM::Role Wed Apr 01 2020 13:36:43 GMT+0100 (British Summer Time)
UPDATE_COMPLETE UnauthRole AWS::IAM::Role Wed Apr 01 2020 13:36:43 GMT+0100 (British Summer Time)
⠹ Updating resources in the cloud. This may take a few minutes...

UPDATE_IN_PROGRESS authcognito62efae15 AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:44 GMT+0100 (British Summer Time)
⠙ Updating resources in the cloud. This may take a few minutes...

UPDATE_COMPLETE             UserPool                                                         AWS::Cognito::UserPool     Wed Apr 01 2020 13:36:47 GMT+0100 (British Summer Time)
UPDATE_IN_PROGRESS          UserPool                                                         AWS::Cognito::UserPool     Wed Apr 01 2020 13:36:46 GMT+0100 (British Summer Time)
UPDATE_ROLLBACK_IN_PROGRESS qi-web-platform-xxxxxx-authcognito62efae15-1FGX3HR4IUXCJ AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:44 GMT+0100 (British Summer Time) User Initiated
⠙ Updating resources in the cloud. This may take a few minutes...

UPDATE_ROLLBACK_FAILED qi-web-platform-xxxxxx-authcognito62efae15-1FGX3HR4IUXCJ AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:50 GMT+0100 (British Summer Time) The following resource(s) failed to update: [UserPoolClientLambda].
UPDATE_FAILED          UserPoolClientLambda                                             AWS::Lambda::Function      Wed Apr 01 2020 13:36:50 GMT+0100 (British Summer Time) The runtime parameter of nodejs6.10 is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs12.x) while creating or updating functions. (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException; Request ID: 21c3afbc-098f-4db6-9012-c61fdadacd6f)
UPDATE_IN_PROGRESS     UserPoolClientLambda                                             AWS::Lambda::Function      Wed Apr 01 2020 13:36:49 GMT+0100 (British Summer Time)
⠹ Updating resources in the cloud. This may take a few minutes...

UPDATE_ROLLBACK_FAILED qi-web-platform-xxxxxx AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:55 GMT+0100 (British Summer Time) The following resource(s) failed to update: [authcognito62efae15, apiqiAppSync].
UPDATE_FAILED          authcognito62efae15            AWS::CloudFormation::Stack Wed Apr 01 2020 13:36:54 GMT+0100 (British Summer Time) Embedded stack arn:aws:cloudformation:eu-west-1:612133432723:stack/qi-web-platform-xxxxxx-authcognito62efae15-1FGX3HR4IUXCJ/76d29bd0-458d-11e9-ab98-0664065f2f2e was not successfully updated. Currently in UPDATE_ROLLBACK_FAILED with reason: The following resource(s) failed to update: [UserPoolClientLambda].
⠏ Updating resources in the cloud. This may take a few minutes...Error updating cloudformation stack
⠦ Updating resources in the cloud. This may take a few minutes...

Following resources failed

Resource Name: qi-web-platform-2019031312391-UserPoolClientLambda-1XS1QEQ4BHWIL (AWS::Lambda::Function)
Event Type: update
Reason: The runtime parameter of nodejs6.10 is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs12.x) while creating or updating functions. (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException; Request ID: 21c3afbc-098f-4db6-9012-c61fdadacd6f)
URL: https://console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/arn%3Aaws%3Acloudformation%3Aeu-west-1%3A612133432723%3Astack%2Fqi-web-platform-xxxxxx-authcognito62efae15-1FGX3HR4IUXCJ%2F76d29bd0-458d-11e9-ab98-0664065f2f2e/events


✖ An error occurred when pushing the resources to the cloud

Resource is not in the state stackUpdateComplete
An error occured during the push operation: Resource is not in the state stackUpdateComplete
```

**Comment by chrisfowler** [comment]
I have managed to clear my blockage. This issue seemed to help, but cannot be clear if it was the solution: [Parameters: [CreateAPIKey] do not exist in the template](https://github.com/aws-amplify/amplify-cli/issues/2626). I also deleted the offending Lambda and recreate separately to amplify, using the expected naming/roles/etc.

I still feel that rolling back a Lambda function to an unsupported runtime, is also an issue/bug that should be looked at...maybe.

Thank you for your time on this!
